jerry-core/ecma/operations/ecma-eval.c

```diff
@@ -82,15 +80,19 @@ ecma_op_eval (ecma_string_t *code_p, /**< code string */
 ecma_value_t
 ecma_op_eval_chars_buffer (const lit_utf8_byte_t *code_p, /**< code characters buffer */
                            size_t code_buffer_size, /**< size of the buffer */
-                           bool is_direct, /**< is eval called directly (ECMA-262 v5, 15.1.2.1.1) */
-                           bool is_called_from_strict_mode_code) /**< is eval is called from strict mode code */
+                           uint32_t parse_opts) /**< ecma_parse_opts_t option bits */
 {
 #ifndef JERRY_DISABLE_JS_PARSER
   JERRY_ASSERT (code_p != NULL);
 
   ecma_compiled_code_t *bytecode_data_p;
 
-  bool is_strict_call = (is_direct && is_called_from_strict_mode_code);
+  uint32_t is_strict_call = ECMA_PARSE_STRICT_MODE | ECMA_PARSE_DIRECT_EVAL;
+
+  if ((parse_opts & is_strict_call) != is_strict_call)
```

Comment:
Are you sure this is correct?

Reply:
I think it's an equivalent conversion. In the previous version the `is_strict_call` variable was only true, if both options were true. So it means whether one of the `ECMA_PARSE_STRICT_MODE` or ` ECMA_PARSE_DIRECT_EVAL` flags are not enabled, the script must be executed in non-strict mode.

Reply:
Ok